### ISSUE TITLE
Given a network, provision IPs on it with specific MAC addreses

### ISSUE THREAD

**Issue by fire** [joyent/sdc-adminui#9]
Given a network, provision IPs on it with specific MAC addreses.

For instance on ovh (dedicated host), you can buy ip subnets, however you need to pre-generate them in their admin menu. 

It would be helpful if I was able to prefill the entire subnet with mac addresses rather than use these commands manually:

sdc sdc-napi /nics -X POST -d '{ "vlan_id": 0, "nic_tag": "external", "ip" :"xxx.xxx.xxx.xxx", "primary": true, "owner_uuid": "UUID", "belongs_to_uuid": "UUID", "network_uuid": "UUID", "belongs_to_type": "zone", "mac":"00:00:00:00:00:00"}'

sdc sdc-vmapi /vms/UUID?action=add_nics  -X POST -d '{"macs":"00:00:00:00:00:00"}'

**Comment by fire** [comment]
Is it possible to have a timeline on this?

**Comment by lloydde** [comment]
@fire this doesn't currently correlate to work in progress. Are you able to code up a solution?

**Comment by fire** [comment]
I'm not very familiar with how adminui works internally, so probably not. 

My first guess without knowledge of the code would be some sort of API change and an automated job, then an ui change.

**Comment by fire** [comment]
@lloydde Not sure if I need to mention you.

**Comment by fire** [comment]
```
<rmustacc> iFire: Well, generally in SDC if a NIC is created it needs to belong to something.
<iFire> well I would put it in the network pool
<rmustacc> That's not what a network pool is.
<iFire> I mean the ip to mac address mapping
<iFire> which the NIC grabs a mapping from a pool
<rmustacc> Right, I understand your goal, I'm just trying to explain that there are going to be some problems based on how SDC is implemented.
<iFire> that sounds troublesome
<rmustacc> NICs in SDC are what have both IP addresses and MAC addresses, and they always belong to something, whether that's a server or a vm.
<rmustacc> So based on my memory, which I could be misremembering, you can't associate a mac address with an IP and still have it be in a general provisioning pool.
<iFire> Could you put it in the provisioning page?
<iFire> with that ui searching for an ip
<rmustacc> Do you mean allowing you to specify a MAC address when provisioning?
<iFire> well looking for an ip and then specificing a MAC. Sure.
<rmustacc> It's technically possible, yes.
...
<iFire> https://github.com/joyent/sdc-adminui/blob/9e6843700cd078dfca50adee198c9ad8b7eee098/www/js/views/provision-vm.js#L263 - Probably a change around here. That adds the metadata for the nic?
```
@lloydde

**Comment by askfongjojo** [comment]
Logged https://smartos.org/bugview/ADMINUI-2239 to track this feature request. 

@fire, we don't have an ETA for this at the moment but I do understand the difficulty involved without this feature. I phrased the ticket a little differently from your original issue description based on my understanding of the discussions above and similar challenges I've heard from other users. Please let me know if you think there is anything else I should add to the ticket. Thanks.

**Comment by fire** [comment]
Remember a key aspect is I can't choose random mac address. They are assigned to specific ips.

That is the reason for this procedure's use.

On Dec 8, 2015, at 10:01 PM, askfongjojo <notifications@github.com> wrote:

Closed #9.

**Comment by fire** [comment]
I mean choose the mac address ip pair. The machine is blocked unless they are both on the list.

On Dec 8, 2015, at 10:01 PM, askfongjojo <notifications@github.com> wrote:

Closed #9.

**Comment by askfongjojo** [comment]
Got it. I've updated the https://smartos.org/bugview/ADMINUI-2239 to include the mac address requirement.

**Comment by fire** [comment]
Misread the issue. It looks good.

On Dec 8, 2015, at 10:01 PM, askfongjojo <notifications@github.com> wrote:

Closed #9.
